user.go

```diff
@@ -1,6 +1,54 @@
 package mockoidc
 
+// User defines the expected user values
 type User struct {
-	Username string
-	Email    string
+	ID                string   `json: id,omitempty`
+	Email             string   `json: email,omitempty`
+	Phone             string   `json: phone,omitempty`
+	PreferredUsername string   `json: preferred_username,omitempty`
+	Address           string   `json: address,omitempty`
+	Groups            []string `json: groups,omitempty`
+	EmailVerified     bool     `json: email_verified,omitempty`
+}
+
+func DefaultUser() *User {
+	return &User{
+		Email:             "user@example.com",
+		Phone:             "555-555-1212",
+		PreferredUsername: "Jon Snow",
+		Address:           "123 Main Street, Brooklyn, NY, 11201",
+		Groups:            []string{"Kings", "Northerners", "Bastards", "SnowMen"},
+		EmailVerified:     true,
+	}
+}
+
+func (u User) CloneForScopes(scopes []string) User {
+	var clone User
+
+	clone.ID = u.ID
+	clone.PreferredUsername = u.PreferredUsername
+
+	for _, scope := range scopes {
+		switch scope {
+		case "profile":
+			clone.Address = u.Address
+			clone.Phone = u.Phone
+		case "email":
+			clone.Email = u.Email
+		case "groups":
+			clone.Groups = u.Groups
```

Comment:
Since `Groups` is a slice, this will copy a reference to the original groups, leading to us potential mutating it on the clone.

```suggestion
			clone.Groups = append(make([]string, 0, len(u.Groups)), u.Groups...)
```

That initializes a new zero'd slice with a capacity for the same length as the original then copies the original in.